File: bookstore1/src/models/CartModel.ts
import { BookModel } from "./BookModel";

export class CartModel {
 id?: number;
 userId!: number;
 bookId!: number;
 quantity!: number;
}

export class CartList {
 id!: number;
 userId!: number;
 bookId!: number;
 quantity!: number;
 price!: number;
 base64image!: string;
 name!: string;
}

export class GetCart {
 results!: CartList[];
 totalRecords!: number;
}

File: BookStoreNew/BookStoreNew.Models/Models/CartModel.cs
using BookStoreNew.Models.ViewModels;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookStoreNew.Models.Models
{
    public class CartModel
    {
        public CartModel() { }

        public CartModel(Cart cart)
        {
            Id = cart.Id;
            UserId = cart.Userid;
            BookId = cart.Bookid;
            Quantity = cart.Quantity;
            BookName = cart.Book.Name;
            Price = cart.Book.Price;
        }

        public int Id { get; set; }
        public int UserId { get; set; }
        public int BookId { get; set; }
        public int Quantity { get; set; }
        public string BookName { get; set; }
        public decimal Price { get; set; }
    }

}
